### DISCUSSION TITLE
Proposed link on the front page

### DISCUSSION THREAD

**Discussion by andrewsolomon** [metacpan/metacpan-web#2606] [Ideas]
At a meeting with TPF's marketing team, people floated the idea of putting a link to https://www.perl.org/learn.html on the front page of https://metacpan.org/ in a blurb along the lines of:

“Just getting started with Perl? [Click here](https://www.perl.org/learn.html) for free learning resources from the Perl community.”

Would that be ok?

**Comment by oalders** [comment]
Sure. I think we could make better use of the space on MetaCPAN and this seems like a good start.

**Reply by andrewsolomon** [reply]
Thanks for giving that the go ahead Olaf. Would you mind making that change? I had a look at the code and even tried running it on Docker but realised it will be a while before I get my head around how to make that tweak.

**Comment by oalders** [comment]
Luckily there is no Docker required: https://github.com/metacpan/metacpan-web#installing-manually

Templates are at https://github.com/metacpan/metacpan-web/tree/master/root

**Reply by andrewsolomon** [reply]
Sorry, I've tried and failed.

**Reply by PaulGWebster** [reply]
Where abouts on the the frontpage was you thinking, would it be possible to get a quick screendump + edit with position, I am not excellent at bootstrap but can give it a try, just not sure where I am targeting to put the link!

**Reply by andrewsolomon** [reply]
Thanks Paul - that's very kind.  [This picture](https://user-images.githubusercontent.com/35419/134430038-72aa9ffb-8605-4dab-b596-65760e65b325.png) is what I had in mind, but I'd be happy wherever it winds up.

**Reply by PaulGWebster** [reply]
ah that makes it a touch easier I was worried you wanted it in the white space between the logo and either the left or right edge :) night time for me, I will give it an edit tomorrow should be fairly easy

**Reply by davorg** [reply]
Google doesn't really like links that are labelled "click here". I suggest the wording should be something more like:

> Just getting started with Perl? Try these [free learning resources] from the Perl community."

(where the link is denoted by [...])

**Reply by PaulGWebster** [reply]
Ok dokey, seem to have got it now just need to build or get it to run locally to prove the changes here is what it will look like: https://i.imgur.com/psFplUV.png just hope I can actually get the stack to run, as the docker appears to be having some issues: https://github.com/metacpan/metacpan-docker/issues/89 and so does running it locally:

```
➜  metacpan-web git:(master) ✗ carton install
➜  metacpan-web git:(master) ✗ yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
```

```
diff --git a/root/base.tx b/root/base.tx
index 71edd239..d81e3a2d 100644
--- a/root/base.tx
+++ b/root/base.tx
@@ -188,6 +188,7 @@
                             </div>
                         </div>
                     </form>
+                   <p class="helper_blurb">Just getting started with perl? Click <a target="_blank" rel="noopener" href="https://www.perl.org/learn.html">here</a> for free learning resources from the Perl community</p>
                 </div>
             </div>
         %%  }
diff --git a/root/static/less/home.less b/root/static/less/home.less
index 8098e5ed..e7a7d7d8 100644
--- a/root/static/less/home.less
+++ b/root/static/less/home.less
@@ -5,6 +5,11 @@
         font-size: 1.2em;
     }

+    .helper_blurb {
+        font-weight: bold;
+        width: 480px;
+    }
+
```

**Reply by PaulGWebster** [reply]
oops let me adjust the wording for free learning resources

**Reply by PaulGWebster** [reply]
That gives us, https://i.imgur.com/MQIHfje.png

and the diff: 
```
diff --git a/root/base.tx b/root/base.tx
index 71edd239..cec1cc26 100644
--- a/root/base.tx
+++ b/root/base.tx
@@ -188,6 +188,7 @@
                             </div>
                         </div>
                     </form>
+                   <p class="helper_blurb">Just getting started with perl? Try these <a target="_blank" rel="noopener" href="https://www.perl.org/learn.html">free learning resources</a> from the Perl community</p>
                 </div>
             </div>
         %%  }
diff --git a/root/static/less/home.less b/root/static/less/home.less
index 8098e5ed..e7a7d7d8 100644
--- a/root/static/less/home.less
+++ b/root/static/less/home.less
@@ -5,6 +5,11 @@
         font-size: 1.2em;
     }

+    .helper_blurb {
+        font-weight: bold;
+        width: 480px;
+    }
+
     #metacpan_search-input {
         width: 400px;
         margin: 0px;

**Reply by davorg** [reply]
Hi @PaulGWebster,

Thanks for taking this one. It's looking good. One small comment.

The square brackets in my example were just there to show where the link goes. They shouldn't be in the actual text.

Cheers,

Dave...

**Reply by PaulGWebster** [reply]
Ah was unsure! I am not up on all the latest UI trends so I assumed it was so it would stand out more as a link :) let me change it

**Reply by PaulGWebster** [reply]
did not see a point of reposting basically the same screenshot etc, so I removed the [ ] in the previous diff

**Reply by PaulGWebster** [reply]
so yes, I can post it as an untested pr (due to issues mentioned at the start of this thread) or just leave it as diffs in this thread or if you have any idea how to get either the docker or local build working I can test it then raise it as a tested PR, what would you like me to do :)

**Reply by davorg** [reply]
Can you push your changes to a GitHub repo (under your account, I guess). I'll download it and try to get it working (probably later this evening).

**Reply by PaulGWebster** [reply]
sure thing here you are https://github.com/OpusVL/metacpan-web

**Comment by oalders** [comment]
> ➜  metacpan-web git:(master) ✗ carton install
➜  metacpan-web git:(master) ✗ yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'

```bash
$ yarn install
yarn install v1.22.11
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

✨  Done in 8.94s.
```

@PaulGWebster which version of `yarn` do you have installed?

**Reply by PaulGWebster** [reply]
```
➜  metacpan-web git:(master) ✗ yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
➜  metacpan-web git:(master) ✗ yarn --version
0.32+git
```

**Reply by PaulGWebster** [reply]
I just did 'apt install apt nodejs' let me see if I can get a later version

**Reply by PaulGWebster** [reply]
```
➜  metacpan-web git:(master) ✗ yarn install
yarn install v1.22.5
[1/4] Resolving packages...
success Already up-to-date.
```

That helped :)

**Reply by PaulGWebster** [reply]
Ok dokey I got the non-docker version working, so here is a PR https://github.com/metacpan/metacpan-web/pull/2607 :) 

Please double and triple check this before sending to production! ideally get someone who knows 'less' to check it, it did however add the text you wish in the right place, so ... if not the right 'lessc' command args its very close :)
